k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageContainerView.java

```diff
@@ -388,7 +389,10 @@ public void displayMessageViewContainer(MessageViewInfo messageViewInfo,
             mSavedState = null;
         }
 
-        String textToDisplay = messageViewInfo.text;
+        String textToDisplay = K9.displayAsPlaintText()
+                ? HtmlConverter.htmlToText(messageViewInfo.text)
```

Comment:
This is not what we want to show. The text displayed here should be the `text` field from `ViewableExtractedText`, please see `MessageViewInfoExtractor` for the relevant code.